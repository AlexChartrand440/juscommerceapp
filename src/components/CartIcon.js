import React from 'react';
import styled from 'styled-components';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { Badge } from 'react-native-paper';

const Wrapper = styled.TouchableOpacity`
  width: 25;
  height: 25;
  right: 16;
`;

const Counter = styled(Badge)`
  position: absolute;
  top: -10;
  right: -10;
  background-color: black;
`;

const CartIcon = ({ cart }) => (
  <Wrapper onPress={() => cart.setVisible(!cart.visible)}>
    <Icon name='shopping-cart' color='#FFF' size={25} />
    <Counter size={18}>{cart.quantity}</Counter>
  </Wrapper>
);

export default CartIcon;
