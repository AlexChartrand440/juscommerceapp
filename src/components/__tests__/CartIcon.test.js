import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import CartIcon from '../CartIcon';

const emptyCart = {
  list: [],
  setList: jest.fn()
};

const oneItemCart = {
  list: [
    {
      product: {
        id: 1,
        name: 'test product1',
        description: 'test description 1',
        image:
          'https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/09/12/11/naturo-monkey-selfie.jpg?w968h681',
        price: 123
      },
      amount: 10
    }
  ],
  setList: jest.fn()
};

test('test empty cart icon', () => {
  const wrapper = renderer.create(<CartIcon cart={emptyCart} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

test('test one item cart', () => {
  const wrapper = renderer.create(<CartIcon cart={oneItemCart} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
