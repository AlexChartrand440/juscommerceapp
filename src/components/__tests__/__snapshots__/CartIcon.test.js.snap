// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test empty cart icon 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 25,
      "opacity": 1,
      "right": 16,
      "width": 25,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "black",
        "borderRadius": 9,
        "color": "#ffffff",
        "fontFamily": "Helvetica Neue",
        "fontSize": 9,
        "height": 18,
        "lineHeight": 18,
        "minWidth": 18,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 4,
        "position": "absolute",
        "right": -10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "top": -10,
      }
    }
  />
</View>
`;

exports[`test one item cart 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 25,
      "opacity": 1,
      "right": 16,
      "width": 25,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "black",
        "borderRadius": 9,
        "color": "#ffffff",
        "fontFamily": "Helvetica Neue",
        "fontSize": 9,
        "height": 18,
        "lineHeight": 18,
        "minWidth": 18,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 4,
        "position": "absolute",
        "right": -10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "top": -10,
      }
    }
  />
</View>
`;
